lib/countSyllables.js

```diff
@@ -0,0 +1,16 @@
+const toSyllabicSymbols = require( './toSyllabicSymbols' )
+
+/**
+ * Calculates the number of syllables according to Sanskrit prosody, Pingala, Matra/Meter/Morae
+ * @param {String} text The string to analyze
+ * @return {number} An integer adding up all the 1's (laghu/light/short) and 2's (guru/heavy/long).
+ * @example
+ * countSyllables( 'ਪ੍ਰਭੂ ਪ੍ਰੇਮੀ ਪੜ੍ਹ ਚੜ੍ਹ ਦ੍ਵੈਤ' )
+ * // expected output: 14
+ */
+const countSyllables = ( text ) => toSyllabicSymbols( text )
+  .split( '' )
+  .map( Number )
+  .reduce( ( a, b ) => a + b )
```

Comment:
If you're typecasting above because you found that it was appending a garbage string, that's because you did not supply an initial value to the reducer, so on first run, it will be `undefined`.

```suggestion
  .reduce( ( a, b ) => a + b, 0 )
```

Reply:
I get the following when trying your suggestion:

```
14) countSyllables()
       should convert 'ਦੌੜ' to '3':
     AssertionError: expected '021' to equal 3
```

Reply:
I can do this to typecast inline, if you prefer (personally I find this more confusing):

```
const countSyllables = ( text ) => toSyllabicSymbols( text )
  .split( '' )
  .reduce( ( a, b ) => +a + +b )
```

Otherwise, as you can see it adds a 0 to the string concatenation process instead of adding them up as numbers.

Reply:
The correct typecast would be for what you wrote would be:
=> `.reduce( ( a, b ) => +a + b )` - a is undefined or null here, and b is typecasted

But I recommend one of the following or a mixture for simplicity:
=> `.reduce( ( a, b ) => b + a )` - b i s a number here, and a is automatically typecasted since it is appended with `+`
=> `.reduce( ( a, b ) => a + b, 0 )` - a is 0 or a number here, and `+ b` forces a typecast

The reason I nitpicked this is because for a long string, you'll be adding an entire unnecessary iteration to the time taken, since a `.map` will have to go over the entire string. Sometimes the clarity can be worth it, but in the case, I don't believe so.

Reply:
That's a fair reason, still getting used to big O complexity

These suggestions don't work (though your logic makes sense to me!):
* `.reduce( ( a, b ) => a + b, 0 )` as shared in a comment above the 0 does not force a typecast apparently
* `.reduce( ( a, b ) => b + a )` i thought this was most clever, but didn't work (even if adding `, 0`)
* `.reduce( ( a, b ) => +a + b )`

This does work:
* `.map(Number).reduce( ( a, b ) => a + b )`
* `.reduce( ( a, b ) => +a + +b) // using plus notation to typecast`

Reply:
does:
- `.reduce( ( a, b ) => +b + a )`
- `.reduce( ( a, b ) => +b + a, 0 )`

work?

Out of curiosity, can you log what `a` and `b` are on the first iteration, with their respective types? using `typeof`. I don't see why my suggestions don't work, unless my initial assumptions of what type they are are wrong.

Re big O, technically this is still O(n), since you're doing O(n + n), but in practice, an extra iteration would definitely have an impact on a long string/text, or if you're doing batch processing.

Reply:
First one, no. Second one, yes!

Here is `console.log( a, typeof ( a ), b, typeof ( b ) )`:
![image](https://user-images.githubusercontent.com/14130567/96310279-159d1980-0fd5-11eb-9b5e-c1670b4fff4f.png)